storagenode/gracefulexit/worker.go

```diff
@@ -169,6 +122,78 @@ func (worker *Worker) Run(ctx context.Context, done func()) (err error) {
 	return errs.Wrap(err)
 }
 
+func (worker *Worker) transferPiece(ctx context.Context, transferPiece *pb.TransferPiece, c pb.SatelliteGracefulExit_ProcessClient) error {
+	pieceID := transferPiece.OriginalPieceId
+	reader, err := worker.store.Reader(ctx, worker.satelliteID, pieceID)
+	if err != nil {
+		transferErr := pb.TransferFailed_UNKNOWN
+		if errs.Is(err, os.ErrNotExist) {
+			transferErr = pb.TransferFailed_NOT_FOUND
+		}
+		worker.log.Error("failed to get piece reader.", zap.Stringer("satellite ID", worker.satelliteID), zap.Stringer("piece ID", pieceID), zap.Error(errs.Wrap(err)))
+		worker.handleFailure(ctx, transferErr, pieceID, c.Send)
+		return err
+	}
+
+	addrLimit := transferPiece.GetAddressedOrderLimit()
+	pk := transferPiece.PrivateKey
+
+	originalHash, originalOrderLimit, err := worker.getHashAndLimit(ctx, reader, addrLimit.GetLimit())
+	if err != nil {
+		worker.log.Error("failed to get piece hash and order limit.", zap.Stringer("satellite ID", worker.satelliteID), zap.Stringer("piece ID", pieceID), zap.Error(errs.Wrap(err)))
+		worker.handleFailure(ctx, pb.TransferFailed_UNKNOWN, pieceID, c.Send)
+		return err
+	}
+
+	putCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	// TODO what's the typical expiration setting?
```

Comment:
not sure but it's okay for now because it didn't look like expiration even got used inside PutPiece right?

Reply:
Yeah it isn't used afaik